src/core/db/repository/category.py

```diff
@@ -22,3 +22,18 @@ async def get_unarchived_subcategories(self, parent_id: int) -> list[Category]:
             select(Category).where(Category.is_archived == False).where(Category.parent_id == parent_id)  # noqa
         )
         return categories
+
+    async def get_unarchived_parents_with_children_count(self):
+        subq = (
+            select(Category.parent_id, func.count(Category.id).label("children_count"))
+            .where(Category.is_archived == False)  # noqa
+            .where(Category.parent_id != None)  # noqa
```

Comment:
Тут можно воспользоваться [функциями sqlalchemy](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#column-element-foundational-constructors) чтобы не пришлось прописывать `# noqa`:
```suggestion
            .where(Category.is_archived == false())
            .where(Category.parent_id != null())
```

Reply:
Отлично! Спасибо.